components/sysutils/nut/Makefile

```diff
@@ -10,44 +10,88 @@
 #
 
 #
-# Copyright 2016 Jim Klimov
+# Copyright 2016-2023 Jim Klimov
 #
 
 BUILD_BITS= 64
+
+# More details on python versions handled here are commented below
+SINGLE_PYTHON_VERSION=	no
```

Comment:
Why is this needed?  I believe the main problem with this `Makefile` is that you try to fiddle with Python versions.  This is almost never needed and is usually counterproductive.  Also, introducing dependency on Python 2.7 is wrong.  Please do not try to do it.  There is only one package that is allowed to depend on Python 2.7: `gimp`.  Python 2.7 was [EOLed almost 4 years ago](https://devguide.python.org/versions/#unsupported-versions).

Reply:
I think this got "lost" in discussion on IRC, so rehashing here :)

On one hand, NUT tries to be as backwards compatible as feasible. Literally systems from turn of millennium like Solaris 8 can still build it. As long as they run (and someone goes through the hassle of setting up a build environment to test it), they may be power-protected.

The python bindings are one aspect of that, with the same PyNUT.py module and its test script re-coded in a few places to survive with both python 2.6/2.7 series and the many 3.x series (half a dozen tried).

There is also an interactive client GUI, NUT-Monitor, which was "forked" and exists as either a Py2+GTK2 or a Py3+QT5 app, and a wrapper script to pick the preferred one by default. From what I remember about the conversion discussion, the Py3 version was less functional about something, can't pinpoint what, so if the user's system has python 2.x - that UI version is the default.

The dependency here is "soft", at least for the end-user installations: *IFF* py2.7 etc. is present (not all installations are new every time), the appropriate version of the app and module would be installed. I suppose (but could not check fully due to the mix of publishers locally) that if the user asks to explicitly install the UI client variant "py2gtk", that would pull in python 2.7 and some of its modules (gtk etc.) but this is not default behaviour.

Reply:
Please behave here as if there is no Python 2.7 available.  If there is some missing functionality (or some other issue) in NUT when compiled with Python 3.x, then the upstream should fix it.  All Pythons older than 3.8 are dead horses (or dead snakes, to be precise :-)).

Reply:
Oh well, at least that was an interesting challenge :D